### PR TITLE
VZ-7127 Separate VPO Webhooks into separate deployment

### PR DESCRIPTION
**Pull request by bryantidd**
Separate VPO webhooks into their own deployment so that they can be scaled and resiliency provided

### CODE REVIEW

File: platform-operator/main.go [hunk 1]
```diff
@@ -173,19 +173,65 @@ func main() {
...
+				os.Exit(1)
+			}
+			if err = (&installv1beta1.Verrazzano{}).SetupWebhookWithManager(mgr, log); err != nil {
+				log.Errorf("Failed to setup install.v1beta1.Verrazzano webhook with manager: %v", err)
+				os.Exit(1)
+			}
+			mgr.GetWebhookServer().CertDir = config.CertDir
+		}
+		// Set up the validation webhook for VMC
+		if config.WebhooksEnabled {
```
Comment (bobdonat) [inline platform-operator/main.go:218]: No need for this if.  It can be included in previous if statement

File: platform-operator/main.go [hunk 2]
```diff
@@ -5,6 +5,7 @@ package main
 
 import (
 	"flag"
+	"github.com/verrazzano/verrazzano/platform-operator/controllers/verrazzano/validator"
```
Comment (bobdonat) [inline platform-operator/main.go:8]: Is this same main.go being used in both the VPO deployment and webhook deployment

Comment (bobdonat) [reply platform-operator/main.go:8]: If that is the case, it appears that a bunch of code is run that is not needed.

Comment (bryantidd) [reply platform-operator/main.go:8]: yes, I would like to improve this in a followup PR.

File: platform-operator/internal/k8s/certificate/certificate.go [hunk 3]
```diff
@@ -162,8 +163,18 @@ func writeFile(filepath string, pem *bytes.Buffer) error {
 
 // UpdateValidatingnWebhookConfiguration sets the CABundle
 func UpdateValidatingnWebhookConfiguration(kubeClient kubernetes.Interface, caCert *bytes.Buffer) error {
+	var oldValidatingWebhook *adminv1.ValidatingWebhookConfiguration
+	oldValidatingWebhook, err := kubeClient.AdmissionregistrationV1().ValidatingWebhookConfigurations().Get(context.TODO(), OldOperatorName, metav1.GetOptions{})
+	if err == nil {
```
Comment (bobdonat) [inline platform-operator/internal/k8s/certificate/certificate.go:168]: This is type of thing we normally do in the upgrade code. Seems to be following a different pattern here.

File: platform-operator/helm_config/charts/verrazzano-platform-operator/templates/deployment.yaml [hunk 4]
```diff
@@ -22,37 +22,20 @@ spec:
...
-              httpHeaders:
-                - name: Content-Type
-                  value: application/json
-            periodSeconds: 5
-            timeoutSeconds: 3
+            exec:
+              command:
+                - /verrazzano/platform-operator/scripts/wait4webhook.sh
+            periodSeconds: 10
+            timeoutSeconds: 30
```
Comment (pfmackin) [inline platform-operator/helm_config/charts/verrazzano-platform-operator/templates/deployment.yaml:38]: We want the probe to retry frequently, maybe every 5 secs.  Also, the webhook might take up to 2 min to start so this has to handle that.

File: platform-operator/internal/k8s/netpolicy/netpolicy.go [hunk 5]
```diff
@@ -32,25 +33,37 @@ const (
...
+	netPolicies := newNetworkPolicies(ip, port)
+	for _, netPolicy := range netPolicies {
+		objKey := &netv1.NetworkPolicy{ObjectMeta: metav1.ObjectMeta{Name: netPolicy.ObjectMeta.Name, Namespace: netPolicy.ObjectMeta.Namespace}}
 
-	opResult, err := controllerutil.CreateOrUpdate(context.TODO(), client, objKey, func() error {
-		netPolicy.Spec.DeepCopyInto(&objKey.Spec)
-		return nil
-	})
+		opResult, err := controllerutil.CreateOrUpdate(context.TODO(), client, objKey, func() error {
+			netPolicy.Spec.DeepCopyInto(&objKey.Spec)
```
Comment (pfmackin) [inline platform-operator/internal/k8s/netpolicy/netpolicy.go:56]: This is not the pattern that we normally use for CreateOrUpdate.  Usually we just provide the object key and set the object data in the callback vs setting the object data before the CreateOrUpdate and doing a deep copy.  This seems like it is ok though.

File: platform-operator/main.go [hunk 6]
```diff
@@ -173,19 +173,65 @@ func main() {
...
+		if err != nil {
+			log.Errorf("Failed to create a controller-runtime manager: %v", err)
+			os.Exit(1)
+		}
+
+		installv1alpha1.SetComponentValidator(validator.ComponentValidatorImpl{})
+		installv1beta1.SetComponentValidator(validator.ComponentValidatorImpl{})
+
+		// Setup the validation webhook
+		if config.WebhooksEnabled {
```
Comment (pfmackin) [inline platform-operator/main.go:205]: All the webhook logic should be put in its own function.  Please do this as a followup

**Review by pfmackin — APPROVED** [review]

**Pull request merged**